src/recordpage.cpp

```diff
@@ -801,9 +813,81 @@ QStringList RecordPage::applicationConfigurations()
 
 void RecordPage::appendOutput(const QString& text)
 {
+    // this regex finds ansi escapes and extracts all the numbers in it
+    QRegularExpression regex(QLatin1String("\\x1b\\[([0-9;]+)m"));
+
     QTextCursor cursor(ui->perfResultsTextEdit->document());
     cursor.movePosition(QTextCursor::End);
-    cursor.insertText(text);
+
+    auto ansiToHtml = [](const QStringList& numbers) -> QString {
+        static bool openColor = false;
+        // only foregroud colors are supported currently
+        const QHash<int, QColor> colors = {
+            {30, Qt::black},       {31, Qt::darkRed},  {32, Qt::darkGreen}, {33, Qt::darkYellow}, {34, Qt::darkBlue},
+            {35, Qt::darkMagenta}, {36, Qt::darkCyan}, {37, Qt::darkGray},  {91, Qt::red},        {92, Qt::green},
+            {93, Qt::yellow},      {94, Qt::blue},     {95, Qt::magenta},   {96, Qt::cyan},       {37, Qt::white}};
+
+        QString tag;
+        for (const auto& number : numbers) {
+            bool ok;
+            int num = number.toInt(&ok);
+
+            if (ok) {
+                if (num == 0) {
+                    if (openColor) {
+                        openColor = false;
+                        tag += QLatin1String("</font>");
+                    }
+                } else if (colors.contains(num)) {
+                    if (openColor) {
+                        tag += QLatin1String("</font>");
+                    }
+
+                    tag += QStringLiteral("<font color=\"%1\">").arg(colors[num].name());
+                    openColor = true;
+                    return tag;
+                }
+            }
+        }
+        return tag;
+    };
+
+    switch (m_outputDecoder) {
+    case Default:
+        cursor.insertText(text);
+        break;
+    case RemoveAnsi: {
+        QString cleanOutput = text;
+        cursor.insertText(cleanOutput.replace(regex, QLatin1String("")));
+        break;
+    }
+    case DecodeAnsi: {
+        // html eats up spaces, replaces them to keep the formating
+        QString text_ = text;
+        text_.replace(QLatin1String(" "), QLatin1String("&nbsp;"));
+        QString colorText;
+        auto matches = regex.globalMatch(text_);
+        if (matches.hasNext()) {
+            auto textList = text_.split(regex);
+            while (matches.hasNext()) {
+                auto match = matches.next();
+
+                if (textList.length() > 1) {
+                    colorText += textList.front() + ansiToHtml(match.captured(1).split(QLatin1String(";")));
```

Comment:
capturedView(1) and https://www.kdab.com/qstringview-diaries-zero-allocation-string-splitting/